third_party/blink/renderer/core/brave_page_graph/page_graph.cc

```diff
@@ -1104,7 +1108,11 @@ void PageGraph::RegisterDocumentNodeCreated(blink::Document* document) {
 
   const String local_tag_name(static_cast<blink::Node*>(document)->nodeName());
   auto* dom_root = AddNode<NodeDOMRoot>(node_id, local_tag_name);
-  dom_root->SetURL(NormalizeUrl(document->Url()).GetString());
+  auto url = NormalizeUrl(document->Url());
+  dom_root->SetURL(url.GetString());
+  if (!source_url_ && url.IsValid() && url.ProtocolIsInHTTPFamily()) {
+    source_url_ = url.GetString();
+  }
```

Comment:
there also `PageGraph::DidCommitLoad` which also updates `NodeDOMRoot` url.

Reply:
I figured we would only need one of those, so put it in `PageGraph::RegisterDocumentNodeCreated`. Do you think we need it there as well?

Reply:
> I figured we would only need one of those, so put it in `PageGraph::RegisterDocumentNodeCreated`. Do you think we need it there as well?

well, if you tried it and it works fine, then I guess it's enough for this exact feature 🙂